key.ui/src/main/java/de/uka/ilkd/key/gui/prooftree/ProofTreeView.java

```diff
@@ -192,6 +193,30 @@ public void updateUI() {
                 }
             }
         };
+        var renderer = delegateView.getCellRenderer() instanceof DefaultTreeCellRenderer
+                ? (DefaultTreeCellRenderer) delegateView.getCellRenderer()
+                : null;
+        delegateView.setCellEditor(new DefaultTreeCellEditor(delegateView, renderer) {
```

Comment:
Ideally, there should be a comment here about the purpose of the anonymous class (it is there to disable editing on non-branching nodes which are not the root, right?).